src/material-experimental/mdc-slider/slider.ts

```diff
@@ -7,12 +7,42 @@
  */
 
 import {
+  BooleanInput,
+  coerceBooleanProperty,
+  coerceNumberProperty,
+  NumberInput
+} from '@angular/cdk/coercion';
+import {Platform} from '@angular/cdk/platform';
+import {DOCUMENT} from '@angular/common';
+import {
+  AfterViewInit,
   ChangeDetectionStrategy,
+  ChangeDetectorRef,
   Component,
+  ContentChildren,
+  ElementRef,
+  EventEmitter,
+  Inject,
+  Input,
+  OnDestroy,
+  Output,
+  QueryList,
+  ViewChild,
+  ViewChildren,
   ViewEncapsulation,
 } from '@angular/core';
-import {MDCSliderFoundation} from '@material/slider';
+import {MDCSliderFoundation, Thumb, TickMark} from '@material/slider';
 import {SliderAdapter} from './slider-adapter';
+import {MatSliderThumb} from './slider-thumb';
+
+/** Represents an interaction event emitted by the MatSlider component. */
+export interface MatSliderThumbInteractionEvent {
```

Comment:
I would still move this to its own file like `slider-events.ts`. It would also be good to expand the JsDoc to explain what specifically is covered by "interaction"

Reply:
I made this change but it creates a circular dependency